pkg/plugins/plugins.go

```diff
@@ -0,0 +1,86 @@
+package plugins
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+)
+
+var (
+	ErrNotImplements = errors.New("Plugin not implements particular driver")
+)
+
+type plugins struct {
+	sync.Mutex
+	plugins map[string]*Plugin
+}
+
+var storage = plugins{plugins: make(map[string]*Plugin)}
+
```

Comment:
@calavera @tiborvass Have integrated this `plugins` mechanism with a proposed modification under https://github.com/mavenugo/docker/commits/libnetwork-plugin-integ.

The proposed changes to this PR : https://github.com/mavenugo/docker/commit/7fc16f40de9c86ef89658043f19e702aff7c25de It essentially brings back the `Handle()` API back into plugins.go

With this changes & the libnetwork integration (as seen in https://github.com/mavenugo/docker/commits/libnetwork-plugin-integ), am able to get `remote` `Network-Driver` integrated with Plugins.

Can you please include the 'Handle()' API back ?